drivers/i2c/i2c_ll_stm32.c

```diff
@@ -115,7 +115,9 @@ int i2c_stm32_runtime_configure(const struct device *dev, uint32_t config)
 #endif
 
 	LL_I2C_Disable(i2c);
+#if !DT_HAS_COMPAT_STATUS_OKAY(st_stm32u0_i2c)
```

Comment:
Are we going to follow this approach or check if the specific bits are defined in SoC's header file (e.g. `I2C_CR1_SMBDEN`) like we do elsewhere in I2C driver (e.g. `I2C_FLTR_ANOFF`)?

Reply:
Hello @ydamigos ,
Thanks for the suggestion, I think I'll do it this way, it works and I don't have any other constraints with it.

Reply:
I would avoid to introduce a new compatible `st,stm32u0-i2c`, just to exclude code, while there is another way doing it which is already there and it is used in STM32 I2C driver. But @erwango has the last word as the maintainer.

Reply:
Thanks @ydamigos for the proposal. I agree that for now, a new compatible doesn't add much value and is more complex. Let's go with the register presence check for now, we'll still be able to come back on that later if required. Thanks for the proposal, @ydamigos!

Reply:
> I would avoid to introduce a new compatible `st,stm32u0-i2c`, just to exclude code, while there is another way doing it which is already there and it is used in STM32 I2C driver. But @erwango has the last word as the maintainer.

hello @ydamigos 
I think my last post was confusing, I said I tested your suggestion and it worked, so I would use that instead , it's simpler indeed.